infrastructure/router/router.go

```diff
@@ -1,18 +1,23 @@
 package router
 
 import (
-	"academy_bootcamp/interface/controller"
+	"github.com/FernandoGal25/academy-go-q42021/interface/controller"
 
 	"github.com/labstack/echo"
 	"github.com/labstack/echo/middleware"
 )
 
-func NewRouter(e *echo.Echo, c controller.AppController) *echo.Echo {
+/*
+	Initializes the endpoints of the pokemon API
+*/
+func NewRouter(c controller.AppController) *echo.Echo {
+	e := echo.New()
 	e.Use(middleware.Logger())
 	e.Use(middleware.Recover())
 
-	e.GET("/items/:id", func(context echo.Context) error { return c.Item.GetItemById(context) })
-	// e.POST("/items", func(context echo.Context) error { return c.Item.CreateItem(context) })
+	e.GET("/pokemon/:ID", func(context echo.Context) error { return c.Pokemon.ActionGetByID(context) })
+	e.GET("/pokemon", func(context echo.Context) error { return c.Pokemon.ActionGetAll(context) })
+	e.POST("/pokemon/:ID", func(context echo.Context) error { return c.Pokemon.ActionPostByID(context) })
```

Comment:
Use plural for resource in path, and it's a good idea to have them separated by endpoint. Also, it's not common to use CAPS for query params.

```suggestion
	e.GET("/pokemons", func(context echo.Context) error { return c.Pokemon.ActionGetAll(context) })
	
	e.GET("/pokemons/:id", func(context echo.Context) error { return c.Pokemon.ActionGetByID(context) })
	e.POST("/pokemons/:id", func(context echo.Context) error { return c.Pokemon.ActionPostByID(context) })
```